### PR TITLE
Lucianoeli/click card img

### PR DESCRIPTION
**Pull request by lucianoeli**
Funtionality to click a marvel hero and see detailed info

### CODE REVIEW

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/CharacterServicesImpl.kt [hunk 1]
```diff
@@ -22,4 +24,17 @@ class CharacterServicesImpl(private val api: MarvelResquestGenerator = MarvelRes
             }
         }
     }
+
+    fun getCharacterById(id: Int): Observable<Character> {
+        return Observable.create { subscriber ->
+            val callResponse = api.createService(MarvelApi::class.java).getChatacterById(id)
+            val response = callResponse.execute()
+            if (response.isSuccessful) {
+                subscriber.onNext(mapper.transform(response.body()!!.data!!.characters[0]))
```
Comment (priettt) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/CharacterServicesImpl.kt]: Try to avoid !! operator

Comment (lucianoeli) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/CharacterServicesImpl.kt]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/mapper/CharacterAditionalInfoMapperService.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.puzzlebench.clean_marvel_kotlin.data.mapper
+
+import com.puzzlebench.clean_marvel_kotlin.data.service.response.CharacterAditionalInfoResponse
+import com.puzzlebench.clean_marvel_kotlin.domain.model.CharacterAditionalInfo
+
+open class CharacterAditionalInfoMapperService: BaseMapperRepository<CharacterAditionalInfoResponse, CharacterAditionalInfo>{
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/mapper/CharacterAditionalInfoMapperService.kt]: improve to use this class properly

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/response/CharacterAditionalInfoResponse.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.puzzlebench.clean_marvel_kotlin.data.service.response
+
+class CharacterAditionalInfoResponse(
+        //val character: Character,
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/response/CharacterAditionalInfoResponse.kt]: is this necessary? if not, delete it pls

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/domain/model/CharacterAditionalInfo.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.puzzlebench.clean_marvel_kotlin.domain.model
+
+class CharacterAditionalInfo(
+        //val character: Character,
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/domain/model/CharacterAditionalInfo.kt]: same

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharacterDetailView.kt [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        fragmentRef.get()?.activity?.applicationContext?.showToast(error)
+    }
+
+    fun showCharacterDetail(character: Character) {
+        val fragment = fragmentRef.get()
+        if (fragment != null) {
+            fragment.description.text = character.description
+            fragment.name.text = character.name
+
+            val imageUrl = character.thumbnail.path + "." + character.thumbnail.extension
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharacterDetailView.kt]: use kotlin concat pls

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharacterDetailView.kt [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        }
+    }
+
+    fun showToastNetworkError(error: String) {
+        fragmentRef.get()?.activity?.applicationContext?.showToast(error)
+    }
+
+    fun showCharacterDetail(character: Character) {
+        val fragment = fragmentRef.get()
+        if (fragment != null) {
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharacterDetailView.kt]: you can replace this by 
fragment ?: and the if body
in this way, you are using kotlin like java and it is not the idea

File: app/src/main/res/layout/character_detail.xml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        android:contentDescription="@string/img_content_description"
+        android:foregroundGravity="center" />
+
+    <TextView
+        android:id="@+id/name"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignParentTop="true"
+        android:layout_centerHorizontal="true"
+        android:layout_marginTop="20dp"
```
Comment (SartiMau) [inline app/src/main/res/layout/character_detail.xml]: no hardcode pls

File: app/src/main/res/layout/character_detail.xml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        android:layout_alignParentLeft="true"
+        android:layout_alignParentTop="true"
+        android:layout_alignParentEnd="true"
+        android:layout_alignParentRight="true"
+        android:layout_centerHorizontal="true"
+        android:layout_marginStart="0dp"
+        android:layout_marginLeft="0dp"
+        android:layout_marginTop="74dp"
+        android:layout_marginEnd="0dp"
+        android:layout_marginRight="0dp" />
```
Comment (SartiMau) [inline app/src/main/res/layout/character_detail.xml]: no hardcode pls

File: app/src/main/res/layout/character_detail.xml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        android:id="@+id/name"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignParentTop="true"
+        android:layout_centerHorizontal="true"
+        android:layout_marginTop="20dp"
+        android:text="@string/no_description_available" />
+
+    <TextView
+        android:id="@+id/description"
```
Comment (SartiMau) [inline app/src/main/res/layout/character_detail.xml:24]: make an style and use it pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharacterDetailView.kt (hunk 5 above)
Comment (lucianoeli) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharacterDetailView.kt]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharacterDetailView.kt (hunk 6 above)
Comment (lucianoeli) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharacterDetailView.kt]: done

File: app/src/main/res/layout/character_detail.xml (hunk 8 above)
Comment (lucianoeli) [reply app/src/main/res/layout/character_detail.xml]: done

File: app/src/main/res/layout/character_detail.xml (hunk 9 above)
Comment (lucianoeli) [reply app/src/main/res/layout/character_detail.xml:24]: done

File: app/src/main/res/layout/character_detail.xml (hunk 7 above)
Comment (lucianoeli) [reply app/src/main/res/layout/character_detail.xml]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/mapper/CharacterAditionalInfoMapperService.kt (hunk 2 above)
Comment (lucianoeli) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/mapper/CharacterAditionalInfoMapperService.kt]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/response/CharacterAditionalInfoResponse.kt (hunk 3 above)
Comment (lucianoeli) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/response/CharacterAditionalInfoResponse.kt]: done

**Review by priettt — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**